subprojects/docs/src/docs/release/notes.md

```diff
@@ -21,151 +21,32 @@ Switch your build to use Gradle @version@ by updating your wrapper properties:
+However, there are times when only a few input files have changed since the last execution, and you'd like to avoid reprocessing all of the unchanged inputs.
+Tasks only reprocessing out-of-date input files are called _incremental tasks_.
+Up to Gradle 5.4, for implementing an incremental task you would use [`IncrementalTaskInputs`](dsl/org.gradle.api.tasks.incremental.IncrementalTaskInputs.html).
 
-```
-import com.google.common.hash.Hashing
-import com.google.common.io.Files
-import org.gradle.api.DefaultTask
-import org.gradle.api.file.ConfigurableFileCollection
-import org.gradle.api.file.DirectoryProperty
-import org.gradle.api.file.FileType
-import org.gradle.api.tasks.InputFiles
-import org.gradle.api.tasks.OutputDirectory
-import org.gradle.api.tasks.PathSensitive
-import org.gradle.api.tasks.PathSensitivity
-import org.gradle.api.tasks.SkipWhenEmpty
-import org.gradle.api.tasks.TaskAction
-import org.gradle.work.ChangeType
-import org.gradle.work.FileChange
-import org.gradle.work.InputChanges
-
-class Checksum extends DefaultTask {
-    @SkipWhenEmpty
-    @PathSensitive(PathSensitivity.NAME_ONLY)
-    @InputFiles
-    ConfigurableFileCollection sources = project.files()
-
-    @OutputDirectory
-    DirectoryProperty outputDirectory = project.objects.directoryProperty()
-
-    @TaskAction
-    void checksum(InputChanges changes) {
-        File outputDir = outputDirectory.get().asFile
-        if (!changes.incremental) {
-            println("Non-incremental changes - cleaning output directory")
-            outputDir.deleteDir()
-            outputDir.mkdirs()
-        }
-        for(FileChange change : changes.getFileChanges(sources)) {
-            File changedFile = change.file
-            if (change.fileType == FileType.FILE) {
-                switch(change.changeType) {
-                    case ChangeType.ADDED: // fall through
-                    case ChangeType.MODIFIED:
-                        outputFileForInput(change).text = hashFileContents(changedFile)
-                        break
-                    case ChangeType.REMOVED:
-                        deleteStaleOutput(outputFileForInput(change))
-                        break
-                }
-            }
-        }
-    }
-
-    private void deleteStaleOutput( File file) {
-        println("Removing old output ${file.name}")
-        file.delete()
-    }
+Now you can use the new [`InputChanges`](dsl/org.gradle.work.InputChanges.html) API for implementing incremental tasks.
+This API addresses some shortcomings of the old API, first and foremost that it is now possible to query for changes of individual input file properties, instead of receiving the changes for all input file properties at once.
+Additionally, the file type and the normalized path can be queried for each change, and the old outputs of the task are automatically removed on non-incremental execution.
```

Comment:
> This API addresses some shortcomings of the old one:
>
>  * You can query the changes for individual file properties now, rather than receiving all the changes for all the input file properties together.
>  * You can now access the file type and normalized path for each change
>  * The old outputs of the task are automatically removed on non-incremental execution

Not sure about the last one. I don't really understand what is meant by "non-incremental execution".

Reply:
If Gradle decides that it needs to do a full rebuild (e.g. when the output files or an input property changed) then the previous outputs of the task are removed.

incremental execution = some files changed, only the changed files are reported
non-incremental execution = cannot determine which files changed, all files are reported as changed.

How about this:
`When Gradle is unable to determine which input files need to be processed, the old outputs of the task are removed`

Reply:
Addressed in https://github.com/gradle/gradle/pull/8952.